lib/promenade/railtie.rb

```diff
@@ -1,13 +1,16 @@
 require "promenade/setup"
 require "promenade/engine"
-require "promenade/client/rack/collector"
+require "promenade/client/rack/http_request_duration_collector"
+require "promenade/client/rack/http_request_queue_time_collector"
 
 module Promenade
   class Railtie < ::Rails::Railtie
     initializer "promenade.configure_rails_initialization" do
       Promenade.setup
       Rails.application.config.middleware.insert_after ActionDispatch::ShowExceptions,
-        Promenade::Client::Rack::Collector
+        Promenade::Client::Rack::HTTPRequestDurationCollector
+      Rails.application.config.middleware.insert_after ActionDispatch::ShowExceptions,
+        Promenade::Client::Rack::HTTPRequestQueueTimeCollector
```

Comment:
Queue time collector should come _before_ request duration collector in middleware stack.